pkg/experiment/ingester/memdb/head.go

```diff
@@ -173,3 +176,16 @@ func (h *Head) flush(ctx context.Context) (*FlushedHead, error) {
 	}
 	return res, nil
 }
+
+// TODO: move this into the symbolizer package when available
+func HasUnsymbolizedProfiles(symbols *symdb.Symbols) bool {
+	locations := symbols.Locations
+	mappings := symbols.Mappings
+	for _, loc := range locations {
+		if loc.MappingId == 0 || int(loc.MappingId) >= len(mappings) ||
```

Comment:
I think mappingId==0 is a valid mapping as it should be an index into rewritten slice of mappings

Reply:
I think the other case `int(loc.MappingId) >= len(mappings)` is also impossible given our dedupSlice implementation

Reply:
```go
func TestUnsymbolized(t *testing.T) {
	profile := &googlev1.Profile{
		StringTable: []string{"", "a"},
		Function: []*googlev1.Function{
			{Id: 4, Name: 1},
		},
		Mapping: []*googlev1.Mapping{{Id: 239, HasFunctions: true}},
		Location: []*googlev1.Location{
			{Id: 5, MappingId: 239, Line: []*googlev1.Line{{FunctionId: 4, Line: 1}}},
		},
		Sample: []*googlev1.Sample{
			{LocationId: []uint64{5}, Value: []int64{1}},
		},
	}
	symbols := NewPartitionWriter(0, &Config{
		Version: FormatV3,
	})
	symbols.WriteProfileSymbols(profile)
	unsymbolized := HasUnsymbolizedProfiles(symbols.Symbols())
	assert.False(t, unsymbolized)
}
```

Please double check the above profile. It looks like it should not have `__unsymbolized__`